userspace/libsinsp/filterchecks.cpp

```diff
@@ -109,6 +109,256 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
 }
 
 #define STR_MATCH(s) str_match_start(val, sizeof (s) -1, s)
+///////////////////////////////////////////////////////////////////////////////
+// sinsp_filter_check_file implementation
+///////////////////////////////////////////////////////////////////////////////
+const filtercheck_field_info sinsp_filter_check_file_fields[] =
+	{
+		{PT_CHARBUF, EPF_NONE, PF_NA, "file.path", "Path for File-related operation", "For any event type that deals with a file, the path the file syscall is operating on"},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "file.source", "Source path for File-related operation", "For any event type that deals with a file, and specifically for a source and target like mv, cp, etc, the source path the file syscall is operating on"},
+		{PT_CHARBUF, EPF_NONE, PF_NA, "file.target", "Target path for File-related operation", "For any event type that deals with a file, and specifically for a target and target like mv, cp, etc, the target path the file syscall is operating on"},
+};
+
+sinsp_filter_check_file::sinsp_filter_check_file()
+{
+	m_info.m_name = "file";
+	m_info.m_desc = "Every syscall that has a file descriptor in its arguments has these fields set with information related to the file.";
```

Comment:
What I was mentioning earlier, this looks the same as `fd.` making the feature harder to understand

Reply:
Good point--I will reword this to it talks about syscalls that act on filesystem paths instead of fds, to emphasize the difference.